java/mx/cider/orchard/TruncatingStringWriter.java

```diff
@@ -0,0 +1,91 @@
+package mx.cider.orchard;
+
+import java.io.*;
+
+/**
+ * A <code>java.io.StringWriter</code> with limits on both a single write and
+ * the resulting string size. Limits are not guaranteed to be exact, because of
+ * appending ellipsis the final size can be a few characters longer.
+ *
+ * This class is not thread-safe.
+ */
+public class TruncatingStringWriter extends StringWriter {
```

Comment:
Seems something nice to unit-test in isolation

Reply:
Somewhat agree, but I also believe that `test/orchard/print_test.clj` unit tests cover this quite well.

Reply:
That would seem more of a side coverage, in a unit test maintainers can have isolated examples and explicitly going over the edge cases

(and pigging back on the topic today's another PR, not everyone wants to read Java all the time - worth optimizing  for that)

Reply:
Done 👍